cantools/subparsers/list.py

```diff
@@ -1,11 +1,42 @@
 import argparse
+from typing import Union
 
 import cantools
 
+from ..database.can.signal import NamedSignalValue
 from .dump.formatting import signal_tree_string
 
 
-def _print_message(message, indent=''):
+def _format_val(val: Union[float, int, str, NamedSignalValue, None],
+                unit: str,
+                value_format_specifier: str) \
+        -> str:
+    """Returns signal value formatted according to a format specifier
+
+    e.g.
+    - ``_format_val(1.234, 'm', '%.2f')`` results in '1.23 m'
+    - ``_format_val('IAmAnEnum', 'm', '%.2f')`` results in 'IAmAnEnum'
+    - ``_format_val(1.234)`` results in '1.234'
+    """
+    if val is None or \
+       not unit or \
+       isinstance(val, str) or \
+       isinstance(val, NamedSignalValue):
+
+        return f'{{:{value_format_specifier}}}'.format(val)
+
+    return f'{{:{value_format_specifier}}} {unit}'.format(val)
+
+
+def _print_message(message,
+                   indent='',
+                   no_format_specifics=False,
```

Comment:
`no_format_specifics=False` is a bit awkward. It's not really clear what that means, without looking further into the code. You could add type annotations here, too.